src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprintAction.java

```diff
@@ -0,0 +1,94 @@
+package org.jenkinsci.plugins.docker.commons.fingerprint;
+
+import hudson.model.Fingerprint;
+import hudson.model.Run;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.CheckForNull;
+import jenkins.model.RunAction2;
+import org.jenkinsci.plugins.docker.commons.Messages;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * Adds a link, which provides a list of fingerprints with {@link DockerRunPtrFingerprintFacet}s.
+ * @author Oleg Nenashev
+ */
+public class DockerFingerprintAction implements RunAction2 {
+
+    private final Set<String> fingerprintKeys;
+    transient @CheckForNull Run<?, ?> run;
```

Comment:
It is always nonnull except for very temporarily, and in practice no one is going to check for it being null, so this annotation is misleading.